services/tasks-svc/stories/PatientCRUD_test.go

```diff
@@ -0,0 +1,684 @@
+		BedId:       bedId2,
+		Consistency: &createRes2.Consistency,
+	})
+	assert.NoError(t, err)
+
+	//
+	// GetPatientsByWard
+	//
+
+	getRes, err := patientClient.GetPatientsByWard(ctx, &pb.GetPatientsByWardRequest{
+		WardId: wardId,
+	})
+	assert.NoError(t, err)
+
+	assert.Len(t, getRes.Patients, 2)
+
+	patient1 := getRes.Patients[0]
+	patient2 := getRes.Patients[1]
+
+	if patient1.Id != createRes1.Id {
+		patient1 = getRes.Patients[1]
+		patient2 = getRes.Patients[0]
+	}
+
+	assert.Equal(t, createRes1.Id, patient1.Id)
+	assert.Equal(t, &bedId1, patient1.BedId)
+	assert.Equal(t, createReq1.HumanReadableIdentifier, patient1.HumanReadableIdentifier)
+	assert.Equal(t, *createReq1.Notes, patient1.Notes)
+	assert.Equal(t, assRes1.Consistency, patient1.Consistency)
+
+	assert.Equal(t, createRes2.Id, patient2.Id)
+	assert.Equal(t, &bedId2, patient2.BedId)
+	assert.Equal(t, createReq2.HumanReadableIdentifier, patient2.HumanReadableIdentifier)
+	assert.Equal(t, *createReq2.Notes, patient2.Notes)
+	assert.Equal(t, assRes2.Consistency, patient2.Consistency)
+}
+
+func TestGetPatientAssignmentByWard(t *testing.T) {
+	ctx := context.Background()
+	patientClient := patientServiceClient()
+
+	wardId, _ := prepareWard(t, ctx, "")
+
+	suffixMatrix := [][]string{
+		{"1 A", "1 B"}, // Room 1
+		{"2 A"},        // Room 2
+		{},             // Room 3
+	}
+
+	roomIds := make([]string, 0, len(suffixMatrix))
+
+	roomConsistencies := make(map[string]string)
+	bedConsistencies := make(map[string]string)
+	patientConsistencies := make(map[string]string)
+
+	bedsForRoom := make(map[string][]string)
+	patientForBed := make(map[string]string)
+
+	for i, bedSfxs := range suffixMatrix {
+		roomSuffix := strconv.Itoa(i + 1)
+		roomId, roomConsistency := prepareRoom(t, ctx, wardId, roomSuffix)
+		roomIds = append(roomIds, roomId)
+		roomConsistencies[roomId] = roomConsistency
+		bedsForRoom[roomId] = make([]string, 0)
+
+		for _, bedSuffix := range bedSfxs {
+			bedId, bedConsistency := prepareBed(t, ctx, roomId, bedSuffix)
+			bedConsistencies[bedId] = bedConsistency
+			bedsForRoom[roomId] = append(bedsForRoom[roomId], bedId)
+
+			patientId := preparePatient(t, ctx, bedSuffix)
+			res, err := patientClient.AssignBed(ctx, &pb.AssignBedRequest{
+				Id:    patientId,
+				BedId: bedId,
+			})
+			assert.NoError(t, err)
+			patientConsistencies[patientId] = res.Consistency
+			patientForBed[bedId] = patientId
+		}
+	}
+
+	res, err := patientClient.GetPatientAssignmentByWard(ctx, &pb.GetPatientAssignmentByWardRequest{WardId: wardId})
+	assert.NoError(t, err)
+
+	assert.Len(t, res.Rooms, len(suffixMatrix))
+
+	zlog.Debug().Interface("res", res).Msg("res")
+
+	for _, room := range res.Rooms {
+		assert.Contains(t, roomIds, room.Id)
+		assert.Equal(t, roomConsistencies[room.Id], room.Consistency)
+
+		for _, bed := range room.Beds {
+			assert.Contains(t, bedsForRoom[room.Id], bed.Id)
+			assert.Equal(t, bedConsistencies[bed.Id], bed.Consistency)
+
+			if bed.Patient == nil {
```

Comment:
`bed.Patient == nil` should never be the case because for every bed a patient gets created, right?

Reply:
yes, but i'ts not guaranteed by the type system and should therefore be tested